### ISSUE TITLE
JsonWire doesn't support typeLiterals

### ISSUE THREAD

**Issue by maxim-ponomarev** [OpenHFT/Chronicle-Wire#738]
`!type` declaration is not supported in JsonWire see pull request https://github.com/OpenHFT/Chronicle-Wire/pull/795

**Comment by RobAustin** [comment]
should its show this "{\"implClass\":!type net.openhft.chronicle.wire.JSONWireTest}",

**Comment by yevgenp** [comment]
> should its show this "{"implClass":!type net.openhft.chronicle.wire.JSONWireTest}",

That's not a valid JSON. Though my PR is still WiP. I can't get an expected output from @peter-lawrey  for all the tests above.

**Comment by RobAustin** [comment]
I spoke with maxim today and ( I think ) he said this would be the best format for the UI:
````
{
  "@net.openhft.chronicle.wire.JSONWireTest$DtoWithClassReference": {
    "implClass": {
      "@type": "net.openhft.chronicle.wire.JSONWireTest"
    },
    "bool": "false"
  }
}
````

**Comment by RobAustin** [comment]
@maxim-ponomarev told me that it was blocking him, so Ive put together a PR for review, https://github.com/OpenHFT/Chronicle-Wire/pull/795

**Comment by RobAustin** [comment]
@maxim-ponomarev and @yevgenp  please just go ahead and alter the PR if you feel there should be changes to it.

**Comment by RobAustin** [comment]
I see there are 5 test failures, I'll have to look at that later.

<img width="950" alt="image" src="https://github.com/OpenHFT/Chronicle-Wire/assets/6506056/96af4819-e39b-4f6e-b3a7-bf4f81f0674c">

**Comment by RobAustin** [comment]
@Yevgen - please re-review

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.25ea5](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.25ea5), [BOM-2.25ea23](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.25ea23)
